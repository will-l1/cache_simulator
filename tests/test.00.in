1024
65536
2
22
22
stats
