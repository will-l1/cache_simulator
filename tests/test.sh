#!/bin/sh

echo ======================================================
echo ====================== TEST $1 ========================
echo ======================================================
if timeout 10 ./$2/$3 < tests/test.$1.in > tests/test.$1.out; then 
  if diff -w tests/test.$1.out tests/test.$1.expected > /dev/null; then
    echo PASSED
  else
    echo FAILED
    echo ======
    echo ____Your_output____ __Expected_Output___
   #echo +++++++++++++++++++ ++++++++++++++++++++
    diff -y -W 80 tests/test.$1.out tests/test.$1.expected 
    echo =====================
    echo Your Output:
    echo ++++++++++++++++
    cat tests/test.$1.out
    echo =====================
    echo Expected Output:
    echo ++++++++++++++++
    cat tests/test.$1.expected
    exit 1
  fi
elif [ $? -eq 124 ]; then
  echo TIMEOUT
  exit 1
else 
  echo Abnormal program termination: the program crashed
  echo Exit code $?
  exit 1
fi
