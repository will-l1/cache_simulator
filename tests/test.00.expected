Loaded value [0xb9cb17b29e5109d2] @ address 0x00000016
Loaded value [0xb9cb17b29e5109d2] @ address 0x00000016
Cache hits: 1, misses: 1 -- hit rate 50%
